### DISCUSSION TITLE
How to remove horizontal whitespaces from Textbox?

### DISCUSSION THREAD

**Discussion by KhorenGrig** [fabricjs/fabric.js#9394] [Q&A] [answered]
Hi Everyone.

I have [small example in here](https://codesandbox.io/s/nice-wu-sjk894?file=/src/App.js). The question is when Textbox is resized horizontally, equal whitespaces are created from both sides (it's expected, _textAlign_ options is set to _center_ ). So I want to remove both whitespaces from both sides **on 'mouse:down' event** and keep the _text chars_ positions the same as it was before. I tried the _prevWidth_ approach, but that approach changes the text char's positions. What I want is to keep the text char's positions.  

_Initial view_

![image](https://github.com/fabricjs/fabric.js/assets/26667922/05c1f819-b379-4c9e-9786-2ec25d8a5fdd)

_Resized view_

![image](https://github.com/fabricjs/fabric.js/assets/26667922/e4f3104c-0e3a-4dad-a753-272d7539e0ce)

_Expected result: Highlited spaces in red should be removed_

![image](https://github.com/fabricjs/fabric.js/assets/26667922/afaaa515-02fc-41c7-b7b7-c343d94619a7)

**How can I do that?**

Thanks in advance )

**Comment by lh-sn** [comment]
Try this: 
`__lineWidths` to get the desired width.
`__charBounds` to get text position.

**Comment by ShaMan123** [comment, marked as answer]
So you want it to move and not resize? Weird.
Anyways `initDimensions` is the place to override text layout

**Reply by ShaMan123** [reply]
https://codesandbox.io/s/fervent-tereshkova-2mn6qq?file=/src/App.js

Or a different control action handler

**Reply by KhorenGrig** [reply]
@ShaMan123 Thanks for your answer. Changing only Textbox **width** shifts chars to left or right. And from your example, I am not able to resize texts horizontally.   My point is to trim spaces from both sides (during  **mouse:down event**) and keep the text char's position like they were on resized, but not trimmed yet mode. Spaces highlighted in red lines should be removed, but text chars should remain inside the red box. _Like this_ 
![image](https://github.com/fabricjs/fabric.js/assets/26667922/b591ebe8-c173-4d0f-b028-5d05bec23cd8)

**Reply by ShaMan123** [reply]
Then before setting width in the overide get the center point, set the width and reset the center point

**Comment by asturur** [comment]
I think that during text parsing we collect the minimum allowed width in: `textbox.dynamicMinWidth`.
If your text is positioned by center ( originX and originY are set to 'center' ) is enough to set the textbox width to `this.dynamicMinWidth`.

If is not set on the center you will need to use the usual trick to keep the center position fixed.
We can explain that later, please first verify if `dynamicMinWidth` is the simplest solution to your issue
